src/util/aabb2.ts

```diff
@@ -1,24 +1,39 @@
-import { vec2 } from 'gl-matrix'
-
 import { Immutable } from '~/types/immutable'
 
-export type Aabb2 = [vec2, vec2]
+// x1, y1, x2, y2
+export type Aabb2 = [number, number, number, number]
```

Comment:
I kinda wish there was a way to do our nominal typing trick to ensure that you can't use just any 4-tuple as an `Aabb2`.

We might be able to do our union-with-tagged-interface trick, but `Aabb2` being an actual array, the typechecker will think that `myAabb._tagProp` is a valid expression. That said, maybe that's fine? The chances of us accidentally making a 4-element array that _also_ has a property called `_tagProp` with a specific value is basically zero.